slasher/db/kv/spanner_test.go

```diff
@@ -0,0 +1,296 @@
+		sm, err := db.EpochSpansMap(ctx, tt.epoch)
+		if err != nil {
+			t.Fatalf("Failed to get validator span map: %v", err)
+		}
+
+		if sm == nil || !reflect.DeepEqual(sm, tt.spanMap) {
+			t.Fatalf("Get should return validator span map: %v got: %v", tt.spanMap, sm)
+		}
+		s, err := db.EpochSpanByValidatorIndex(ctx, 1, tt.epoch)
+		if err != nil {
+			t.Fatalf("Failed to get validator span for epoch 1: %v", err)
+		}
+		if !reflect.DeepEqual(s, tt.spanMap[1]) {
+			t.Fatalf("Get should return validator spans for epoch 1: %v got: %v", tt.spanMap[1], s)
+		}
+	}
+}
+
+func TestStore_DeleteEpochSpans(t *testing.T) {
+	app := cli.NewApp()
+	set := flag.NewFlagSet("test", 0)
+	db := setupDB(t, cli.NewContext(app, set, nil))
+	defer teardownDB(t, db)
+	ctx := context.Background()
+
+	for _, tt := range spanTests {
+		err := db.SaveEpochSpansMap(ctx, tt.epoch, tt.spanMap)
+		if err != nil {
+			t.Fatalf("Save validator span map failed: %v", err)
+		}
+	}
+
+	for _, tt := range spanTests {
+		sm, err := db.EpochSpansMap(ctx, tt.epoch)
+		if err != nil {
+			t.Fatalf("Failed to get validator span map: %v", err)
+		}
+		if sm == nil || !reflect.DeepEqual(sm, tt.spanMap) {
+			t.Fatalf("Get should return validator span map: %v got: %v", tt.spanMap, sm)
+		}
+		err = db.DeleteEpochSpans(ctx, tt.epoch)
+		if err != nil {
+			t.Fatalf("Delete validator span map error: %v", err)
+		}
+		sm, err = db.EpochSpansMap(ctx, tt.epoch)
+		if err != nil {
+			t.Fatal(err)
+		}
+		if !reflect.DeepEqual(sm, map[uint64]types.Span{}) {
+			t.Errorf("Expected validator span map to be deleted, received: %v", sm)
+		}
+	}
+}
+
+func TestValidatorSpanMap_DeleteWithCache(t *testing.T) {
+	app := cli.NewApp()
+	set := flag.NewFlagSet("test", 0)
+	set.Bool(flags.UseSpanCacheFlag.Name, true, "enable span map cache")
+	db := setupDB(t, cli.NewContext(app, set, nil))
+	defer teardownDB(t, db)
+	ctx := context.Background()
+
+	for _, tt := range spanTests {
+		err := db.SaveEpochSpansMap(ctx, tt.epoch, tt.spanMap)
+		if err != nil {
+			t.Fatalf("Save validator span map failed: %v", err)
+		}
+	}
+	// wait for value to pass through cache buffers
+	time.Sleep(time.Millisecond * 10)
+	for _, tt := range spanTests {
+		sm, err := db.EpochSpansMap(ctx, tt.epoch)
+		if err != nil {
+			t.Fatalf("Failed to get validator span map: %v", err)
+		}
+		if sm == nil || !reflect.DeepEqual(sm, tt.spanMap) {
+			t.Fatalf("Get should return validator span map: %v got: %v", tt.spanMap, sm)
+		}
+		err = db.DeleteEpochSpans(ctx, tt.epoch)
+		if err != nil {
+			t.Fatalf("Delete validator span map error: %v", err)
+		}
+		// wait for value to pass through cache buffers
+		time.Sleep(time.Millisecond * 10)
+		sm, err = db.EpochSpansMap(ctx, tt.epoch)
+		if err != nil {
+			t.Fatal(err)
+		}
+		if !reflect.DeepEqual(sm, map[uint64]types.Span{}) {
+			t.Errorf("Expected validator span map to be deleted, received: %v", sm)
+		}
+	}
+}
+
+//func TestValidatorSpanMap_SaveOnEvict(t *testing.T) {
```

Comment:
Is there a reason this test is commented out?

Reply:
wasn't sure about sizing of cache. will bring it back in

Reply:
thanks